src/models/event-timeline-set.ts

```diff
@@ -56,6 +56,38 @@ export interface IRoomTimelineData {
     liveEvent?: boolean;
 }
 
+export interface IAddLiveEventOptions {
+    /** Applies to events in the timeline only. If this is 'replace' then if a
+     * duplicate is encountered, the event passed to this function will replace
+     * the existing event in the timeline. If this is not specified, or is
+     * 'ignore', then the event passed to this function will be ignored
+     * entirely, preserving the existing event in the timeline. Events are
+     * identical based on their event ID <b>only</b>. */
+    duplicateStrategy?: DuplicateStrategy;
+    /** Whether the sync response came from cache */
+    fromCache?: boolean;
+    /** The state events to reconcile metadata from */
+    roomState?: RoomState;
+    /** Whether the state is part of the first state snapshot we're seeing in
+     *  the room. This could be happen in a variety of cases:
+     *  1. From the initial sync
+     *  2. It's the first state we're seeing after joining the room
+     *  3. Or whether it's coming from `syncFromCache` */
+    fromInitialState?: boolean;
+}
```

Comment:
Instead of dealing with unwieldy long function signatures with unlabeled and optional arguments in the usage, I've refactored some of these into option objects now that I am adding yet another `boolean` here.

Before:

```js
room.addLiveEvents(mainTimelineEvents, null, fromCache, timelineWasEmpty);
```

After:

```js
room.addLiveEvents(mainTimelineEvents, {
    duplicateStrategy: null,
    fromCache,
    fromInitialState: timelineWasEmpty
});
```